diagnostic.md

```diff
@@ -5,74 +5,86 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```md
-// your response here
+The back end is the the database or model in MVC terminology. Without it, your
+would not be able to save or commit any data to the back end data store or base. Think of Amazon.com that is not attached to a datbase. Not possible to function.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```md
-// your response here
+The model is reponsible for fetching the data. A Model directly manages the data in our application, and provides a representation of that data for the rest of the application to use.
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```md
-// your response here
+Only the controller
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```md
-// your response here
+In API-style applications, serializers replace all the views that we care about so we can safely skip the views as well.
 ```
 
 What does C.R.U.D stand for?
 
 ```md
-// your response here
+Create
+Read
+Update
+Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```md
-// your response here
+Model
 ```
 
 List at least 5 standard rails actions that C.R.U.D requests correspond to?
 
 ```md
-// your response here
+INDEX, SHOW, CREATE, UPDATE, DESTROY
 ```
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```md
-// your response here
+1. The URL for the get message in the view is passed to the controller.
+2. The controller then pass this message off to the model.
+  2a. First it checks the routes to ensure it has an INDEX.
+  2b. Then it checks the controller to  invoke the necessary method to render the jsoon.
+3. Then the model takes over an rails converts the string to a SQL command that is then run on the back end.
+4. Then that data, if it exists, is returned to the controller who then returns it to the view to display the output in the HTML page.
 ```
 
 What is the command to generate a new rails-api app?
-
-```bash
-// your response here
+```
+ bin/rails generate
 ```
 
 What is the command to start an instance of a rails server?
 
 ```bash
-// your response here
+ bin/rails server
 ```
 
 What are the commands to drop, create, migrate and seed a database from the command
 line? (5 bullet points)
 
 ```bash
-// your response here
+bin/rails drop
+bin/rake db:create
+bin/rake db:migrate
+bin/rake db:seed
+bin/rake db:examples
 ```
 
 What is the command to scaffold a pet with a name and age attributes (hint:
 Also think of the data types for each attribute)?
 
 ```bash
-// your response here
+bin/rails generate model Pet name:string age:integer
```

Comment:
Super close but we aren't generating a model only, we want to scaffold all of the different parts of pet:

`rails generate scaffold pet name:string age:integer`